examples/proc_GDS_pulse_capture.py

```diff
@@ -0,0 +1,121 @@
+"""
+Calculate actual beta as a function of pulse length
+===================================================
+Assuming the data is the capture of the pulse sequence as seen on the GDS
+oscilloscope, here the data is converted to analytic power, frequency filtered
+and the absolute is taken prior to integrating to return the beta value (us
+sqrt(W)). 
+"""
+import pyspecdata as psd
+import matplotlib.pyplot as plt
+from numpy import sqrt, array
+
+show_all = False
+with psd.figlist_var() as fl:
+    for filename, nodename, beta90_range, beta180_range in [
+        (
+            "240729_test_p90_pulse_calib.h5",
+            "pulse_calib_2",
+            (None, 34e-6),
+            (34e-6, None),
+        ),
+    ]:
+        d = psd.nddata_hdf5(
+            filename + f"/{nodename}",
+            directory=psd.getDATADIR(exp_type="ODNP_NMR_comp/test_equipment"),
+        )
+        beta_1 = []
+        beta_2 = []
+        for j in range(len(d["p_90"])):
+            s = d["p_90", j].C
+            s.set_units("t", "s")
+            s *= 101.35  # attenutation ratio
+            s /= sqrt(2)  # Vrms
+            s /= sqrt(50)  # V/sqrt(R) = sqrt(P)
+            if show_all:
+                fl.basename = r"programmed $\beta$ = %s" % str(
+                    s.get_prop("set_beta")[j]
+                )
+                fig, ax_list = plt.subplots(1, 2, figsize=(10, 4))
+                fig.suptitle(fl.basename)
+                # {{{ Plot raw capture
+                ax_list[1].set_ylabel(r"$\sqrt{P_{pulse}}$")
+                ax_list[0].set_ylabel(r"$\sqrt{P_{pulse}}$")
+                ax_list[1].set_ylabel(None)
+                ax_list[1].set_title("180 Pulse")
+                fl.next("Pulse Calibration")
+                fl.plot(s["t":beta90_range], alpha=0.2, color="blue", ax=ax_list[0])
+                fl.plot(s["t":beta180_range], alpha=0.2, color="blue", ax=ax_list[1])
+                # }}}
```

Comment:
again diagnostic with the option set to False as I assume we don't want to see all 50 pulse settings

Reply:
again, don't put things in a PR which belong as code comments.  If you feel that you need a PR comment to explain the code, you're probably missing a code comment